packages/aws-cdk-lib/aws-cloudwatch/test/dashboard.test.ts

```diff
@@ -230,6 +241,178 @@ describe('Dashboard', () => {
   });
+
+  test('dashboard has initial select/pattern variable', () => {
+    // GIVEN
+    const stack = new Stack();
+    // WHEN
+    new Dashboard(stack, 'Dashboard', {
+      variables: [new ValueDashboardVariable({
+        type: VariableType.PATTERN,
+        value: 'us-east-1',
+        inputType: VariableInputType.SELECT,
+        id: 'region3',
+        label: 'RegionPatternWithValues',
+        defaultValue: 'us-east-1',
+        visible: true,
+        values: [{ label: 'IAD', value: 'us-east-1' }, { label: 'DUB', value: 'us-west-2' }],
+      })],
+    });
+
+    // THEN
+    const resources = Template.fromStack(stack).findResources('AWS::CloudWatch::Dashboard');
+    expect(Object.keys(resources).length).toEqual(1);
+    const key = Object.keys(resources)[0];
+    hasVariables(resources[key].Properties, [
+      { defaultValue: 'us-east-1', id: 'region3', inputType: 'select', label: 'RegionPatternWithValues', pattern: 'us-east-1', type: 'pattern', values: [{ label: 'IAD', value: 'us-east-1' }, { label: 'DUB', value: 'us-west-2' }], visible: true },
+    ]);
+
+  });
+
+  test('dashboard has initial and added variable', () => {
+    // GIVEN
+    const stack = new Stack();
+    const dashboard = new Dashboard(stack, 'Dashboard', {
+      variables: [new ValueDashboardVariable({
+        type: VariableType.PATTERN,
+        value: 'us-east-1',
+        inputType: VariableInputType.SELECT,
+        id: 'region3',
+        label: 'RegionPatternWithValues',
+        defaultValue: 'us-east-1',
+        visible: true,
+        values: [{ label: 'IAD', value: 'us-east-1' }, { label: 'DUB', value: 'us-west-2' }],
+      })],
+    });
+
+    dashboard.addVariable(new ValueDashboardVariable({
+      type: VariableType.PATTERN,
+      value: 'us-east-1',
+      inputType: VariableInputType.INPUT,
+      id: 'region2',
+      label: 'RegionPattern',
+      defaultValue: 'us-east-1',
+      visible: true,
+    }));
+
+    // THEN
+    const resources = Template.fromStack(stack).findResources('AWS::CloudWatch::Dashboard');
+    expect(Object.keys(resources).length).toEqual(1);
+    const key = Object.keys(resources)[0];
+    hasVariables(resources[key].Properties, [
+      { defaultValue: 'us-east-1', id: 'region3', inputType: 'select', label: 'RegionPatternWithValues', pattern: 'us-east-1', type: 'pattern', values: [{ label: 'IAD', value: 'us-east-1' }, { label: 'DUB', value: 'us-west-2' }], visible: true },
+      { defaultValue: 'us-east-1', id: 'region2', inputType: 'input', label: 'RegionPattern', pattern: 'us-east-1', type: 'pattern', visible: true },
+    ]);
+
+  });
+
+  test('dashboard has property/select search variable', () => {
+    // GIVEN
+    const stack = new Stack();
+    const dashboard = new Dashboard(stack, 'Dashboard');
+
+    // WHEN
+    dashboard.addVariable(new SearchDashboardVariable({
+      defaultValue: '__FIRST',
```

Comment:
I just realized that `__FIRST` is a special value. Can we make that into a constant so that it can be referenced like:

```ts
defaultValue: `DefaultValue.FIRST`
```

Reply:
so just a class with a static constant?

Reply:
WDYT? Maybe its unnecessary.

Reply:
Well, I made it take a bit more responsibility in https://github.com/aws/aws-cdk/pull/26285/commits/d8cc14a9494c9b907248b2437124a7b8557b512a, let me know what you think